Backport of the following patch from git main to 1.11.17.0:

From 8a8c2fd5e6c016eb2d22d2522bdeb9f165f319fa Mon Sep 17 00:00:00 2001
From: Larry Gritz <lg@larrygritz.com>
Date: Mon, 18 Apr 2022 11:45:34 -0700
Subject: [PATCH 1/3] Support for LLVM 14

API changes we had to take into account:
* TargetRegistry.h location
* No more DisableTailCalls field in PassManagerBuilder.

Signed-off-by: Larry Gritz <lg@larrygritz.com>
---
 .github/workflows/ci.yml          | 10 +++++-----
 INSTALL.md                        |  2 +-
 src/build-scripts/build_llvm.bash |  2 +-
 src/liboslexec/llvm_util.cpp      |  7 ++++++-
 4 files changed, 13 insertions(+), 8 deletions(-)

diff --git a/src/liboslexec/llvm_util.cpp b/src/liboslexec/llvm_util.cpp
index b30c89318..22d4134ba 100644
--- a/src/liboslexec/llvm_util.cpp
+++ b/src/liboslexec/llvm_util.cpp
@@ -45,7 +45,11 @@
 #include <llvm/Support/raw_os_ostream.h>
 #include <llvm/IR/LegacyPassManager.h>
 #include <llvm/IR/ValueSymbolTable.h>
+#if OSL_LLVM_VERSION < 140
 #include <llvm/Support/TargetRegistry.h>
+#else
+#include <llvm/MC/TargetRegistry.h>
+#endif
 
 #include <llvm/Bitcode/BitcodeReader.h>
 #include <llvm/Bitcode/BitcodeWriter.h>
